text/0907-pnpm-support.md

```diff
@@ -0,0 +1,153 @@
+  accepted: Fill this in with the URL for the Proposal RFC PR
+project-link: Leave as is
+suite: Leave as is
+-->
+
+# `pnpm` support
+
+## Summary
+
+Enable Ember CLI users to opt into using `pnpm` for package management.
+
+## Motivation
+
+[`pnpm`](https://pnpm.io/) is a popular alternative to both `npm` and `yarn` that prioritizes correctness, especially for `peerDependencies` and monorepos. 
+
+`pnpm` has very active maintainance, support, and financial funding.
+Their [website](https://pnpm.io/) states that `pnpm` is:
+ - Fast: `pnpm` is up to 2x faster than the alternatives 
+ - Efficient: Files inside `node_modules` are cloned or hard linked from a single content-addressable storage 
+ - Supports monorepos: `pnpm` has built-in support for multiple packages in a repository
+ - Strict: `pnpm` creates a non-flat `node_modules` by default, so code has no access to arbitrary packages.
+
+For folks with lots of projects on their computers, `pnpm` is _extremely_ space-efficient.
+Where `npm` and `yarn` would duplicate `node_modules` per-project, `pnpm` will only download a package (at a specific version) once across your whole machine.
+
+Additionally, `npm` and `yarn` repeatedly have demonstrated that their strategies with `peerDependencies` are not correct, and it is vitally important we use and support a tool that can guide folks towards correctly creating addons. For example, `@embroider/macros` relies on node's resolution algorithm, so having `peerDependencies` resolved correctly is important for `dependencySatisfies` to work as expected in monorepos. 
+
+Ember CLI currently only supports `npm` (default) and `yarn` for project initialization as well as various commands. At present, projects work with `pnpm`, but the tooling is unaware.
+
+
+
+## Detailed design
+
+Enabling `pnpm` is designed as opt-in to prevent disruptions to developer's current workflow, much the same as `yarn`.
+
+There are a few integration points that we must support for `pnpm` (and any package manager):
+ - `ember install`
+ - `ember init`, `ember new`, `ember addon`
+ - `ember try:one`, `ember try:each`
+ - generated C.I. configs
+ - Documenation
+
+### `ember install`
+
+There are two mechanisms through which to opt-in.
+The first one is the presence of a `pnpm-lock.yaml` file in the project root.
+
+The `pnpm-lock.yaml` file is generated by `pnpm` when you run `pnpm install` (or the shorter `pnpm i`),
+so we assume that its presence means the developer intends to use `pnpm` to manage their dependencies.
+
+Alternatively you, you can force Ember CLI to use `pnpm` with the `--pnpm` flag, and symmetrically,
+you can force Ember CLI to not use `pnpm` with the `--no-pnpm` flag.
+
+To recap:
+
+- `ember install ember-resources` with `pnpm-lock.yaml` present will use `pnpm`
+- `ember install ember-resources` without `pnpm-lock.yaml` present will use npm
+- `ember install ember-resources --pnpm` will use `pnpm`
+- `ember install ember-resources --no-pnpm` will use npm
```

Comment:
What's the use case for this? If you had a pnpm-lock.yaml and didn't want to use pnpm?